src/Installer.php

```diff
@@ -160,8 +176,51 @@ private function installPuli()
 
         $httpClient = new HttpClient($this->disableTls, $this->cafile);
 
+        $versionUrl = sprintf(
+            static::VERSION_API_URL,
+            $this->disableTls ? 'http' : 'https'
+        );
+
+        $versions = array();
+        for ($retries = 3; $retries > 0; --$retries) {
+            $versions = $this->downloadVersions($httpClient, $versionUrl);
```

Comment:
When you added the exception, you can do:

```php
try {
    $versions = $this->downloadVersions($httpClient, $versionUrl);
} catch (RuntimeException $e) {
    $this->error($e->getMessage);
    continue;
}
```

You can remove the null-check below.